src/traces/bar/attributes.js

```diff
@@ -39,6 +39,51 @@ var marker = extendFlat({
         max: 1,
         editType: 'style',
         description: 'Sets the opacity of the bars.'
+    },
+    patternfill: {
+        shape: {
+            valType: 'enumerated',
+            values: ['', '/', '\\', 'x', '-', '|', '+', '.'],
+            dflt: '',
+            arrayOk: true,
+            editType: 'style',
+            description: [
+                'Sets the shape of the pattern fill.',
+                'By default, no pattern is used for filling the area.',
+            ].join(' ')
+        },
+        bgcolor: {
+            valType: 'color',
+            arrayOk: true,
+            editType: 'style',
+            description: [
+                'Sets the background color of the pattern fill.',
+                'Defaults to a transparent background.',
+            ].join(' ')
+        },
+        scale: {
+            valType: 'number',
+            min: 0,
+            dflt: 1,
+            arrayOk: true,
+            editType: 'style',
+            description: [
+                'Sets the scale of the pattern fill.',
+                'Defaults to 1.',
+            ].join(' ')
+        },
+        solidity: {
+            valType: 'number',
+            min: 0,
+            dflt: 1,
+            arrayOk: true,
+            editType: 'style',
+            description: [
+                'Sets the solidity of the pattern fill.',
+                'Defaults to 1.',
```

Comment:
Again, can we make the scale more meaningful? I guess it would be difficult to get it to exactly match a percentage, but something like 0 to 100, where 0 is only bgcolor and 100 is only foreground color, and we try to make it somewhat linear so that 50% has about the same amount of ink whether it's `x` or `.` pattern, would be nice.

Reply:
This is a very good idea. I implemented it in 39e35ccc02eb5d76ea7bbdc88be61a7cb1943918.
One concern is that, by default, the linewidth of cross hatches (`+`, `x`) becomes the half of other hatches (e.g., `/`), which might not be visually appealing.

Reply:
Very nicely done! Here's your `bar_patternfill` mock with all the sizes and solidities set to default (0.3 and 8):

<img width="1048" alt="Screen Shot 2021-03-01 at 9 47 05 AM" src="https://user-images.githubusercontent.com/2678795/109513447-1c539700-7a73-11eb-9aea-414f2f7eb2cf.png">

And with `size: 20`:

<img width="1041" alt="Screen Shot 2021-03-01 at 9 53 00 AM" src="https://user-images.githubusercontent.com/2678795/109514393-fc70a300-7a73-11eb-8894-72a6cbd4cca0.png">

To my eye the thinner linewidth of crosshatches compared to unidirectional lines is not a problem, and this way the overall visual weight is preserved as you change shape. @nicolaskruchten do you agree?

Reply:
I could be missing something, but I don't see any mechanism in SVG to set the origin of pattern tiling separately for each element being patterned. Anyway if you have two elements with the same pattern touching each other, you'd like them to match origins so the pattern continues smoothly. So all things considered I think we should keep it as is.
![Screen Shot 2021-03-01 at 10 32 45 AM](https://user-images.githubusercontent.com/2678795/109519709-7ce5d280-7a79-11eb-916c-cb5ab699d01a.png)

Reply:
OK, I can live with that.